### ISSUE TITLE
Xonsh chokes on pandas column indexing

### ISSUE THREAD

**Issue by dotsdl** [xonsh/xonsh#4068]
When using `pandas` like the following, I now get an error from Xonsh:
```
import pandas as pd
df = pd.DataFrame({'a': [1, 2, 3], 'b': [4, 5, 6]})

# fails with error below
df[['a', 'b']]
```
with error:
```
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/xonsh/ptk_shell/shell.py", line 238, in _push
    code = self.execer.compile(src, mode="single", glbs=self.ctx, locs=None)
  File "/usr/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 23992, in compile
    tree = self.parse(input, ctx, mode=mode, filename=filename, transform=transform)
  File "/usr/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 23952, in parse
    tree, input = self._parse_ctx_free(input, mode=mode, filename=filename)
  File "/usr/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 24076, in _parse_ctx_free
    tree = self.parser.parse(
  File "/usr/lib/python3.9/site-packages/xonsh/parsers/base.py", line 497, in parse
    tree = self.parser.parse(input=s, lexer=self.lexer, debug=debug_level)
  File "/usr/lib/python3.9/site-packages/xonsh/ply/ply/yacc.py", line 335, in parse
    return self.parseopt_notrack(input, lexer, debug, tracking, tokenfunc)
  File "/usr/lib/python3.9/site-packages/xonsh/ply/ply/yacc.py", line 1122, in parseopt_notrack
    p.callable(pslice)
  File "/usr/lib/python3.9/site-packages/xonsh/parsers/base.py", line 2221, in p_atom_expr
    p[0] = self.apply_trailers(p[1], p[2])
  File "/usr/lib/python3.9/site-packages/xonsh/parsers/base.py", line 2215, in apply_trailers
    assert False
AssertionError

```

This error did not occur with the same usage in a previous release, perhaps under Python 3.8; I can try to pin down last known configuration this worked under.

## xonfig

<details>

```
$ xonfig

+------------------+-----------------+
| xonsh            | 0.9.24          |
| Python           | 3.9.1           |
| PLY              | 3.11            |
| have readline    | True            |
| prompt toolkit   | 3.0.10          |
| shell type       | prompt_toolkit  |
| pygments         | 2.7.4           |
| on posix         | True            |
| on linux         | True            |
| distro           | arch            |
| on darwin        | False           |
| on windows       | False           |
| on cygwin        | False           |
| on msys2         | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
| on jupyter       | False           |
| jupyter kernel   | None            |
+------------------+-----------------+

```

</details>

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by gforsyth** [comment]
Hey @dotsdl -- yeah, you'll want to downgrade to Python 3.8 for `xonsh`, we don't have a handle on all the AST changes yet

**Comment by anki-code** [comment]
Fixed in master by #4101.


Tested in docker:
```python
docker run --rm -it python:3.9-slim /bin/bash -c "apt update && apt install -y git && pip install -U git+https://github.com/xonsh/xonsh.git && xonsh"
```

**Comment by dotsdl** [comment]
Awesome, thank you @anki-code!
